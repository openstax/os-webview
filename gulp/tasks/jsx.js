const path = require('path');
const gulp = require('gulp');
const config = require('../config');
const pi = require('gulp-load-plugins')({
    pattern: ['gulp-*', 'gulp.*', 'del']
});

function jsx() {
    const configSrcApp = `${config.src}/app`;
    const configDestApp = `${config.dest}/app`;

    return gulp.src(`${configSrcApp}/**/*.jsx`, {
        since: gulp.lastRun(jsx)
    })
    .pipe(pi.sourcemaps.init({loadMaps: true}))
    .pipe(pi.rename((uri) => {
        uri.extname = '.jsx.js';
    }))
    .pipe(pi.babel({
        compact: false,
        presets: ['@babel/preset-react']
    }))
    // prefix the sourcemaps with '../src/' so webpack can find them
    .pipe(pi.sourcemaps.mapSources(function(sourcePath, file) {
      const sourcePathHtml = sourcePath.replace('.jsx.js', '.jsx');
      const rel = path.relative(path.dirname(`${configDestApp}/${sourcePath}`), `${configSrcApp}/${sourcePathHtml}`)
      return rel;
    }))
    .pipe(pi.sourcemaps.write('.'))
    .pipe(gulp.dest(`${configDestApp}`));
}


function watchJsx() {
    gulp.watch([`${config.src}/**/*.jsx`], config.watchOpts, gulp.series(
        jsx
    ));
}

exports.jsx = jsx;
exports.jsx.watch = watchJsx;
