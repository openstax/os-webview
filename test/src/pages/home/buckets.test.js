import {makeMountRender, snapshotify} from '../../../helpers/jsx-test-utils.jsx';
import Buckets from '~/pages/home/buckets/buckets.jsx';
import higherEd from '../../data/buckets';

describe('Buckets', () => {
    const data = higherEd.row_3.map((x) => ('value' in x) ? x.value : x);
    const wrapper = makeMountRender(Buckets, {
        bucketModels: data
    })();

    it('matches snapshot', () => {
        expect(snapshotify(wrapper)).toMatchSnapshot();
    });
    it('creates two buckets', () => {
        expect(wrapper.find('.bucket').length).toBe(2);
    });
});
