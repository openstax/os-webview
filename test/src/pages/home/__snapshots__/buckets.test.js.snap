// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buckets matches snapshot 1`] = `"<div class=\\"buckets-section\\"><div class=\\"bucket  left\\"><div class=\\"quote\\"><div><span class=\\"title\\">Our Impact</span><blurb-html class=\\"blurb\\"><p>Faculty at over 3,000 institutions are opening doors for student success by adopting OpenStax.</p></blurb-html></div><a class=\\"link-with-chevron\\" href=\\"http://openstax.org/impact\\">See Our Impact <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"chevron-right\\" class=\\"svg-inline--fa fa-chevron-right fa-w-10 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\"><path fill=\\"currentColor\\" d=\\"M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z\\"></path></svg></a></div></div><div class=\\"bucket  left\\"><div class=\\"quote\\"><div><span class=\\"title\\">OpenStax Partners</span><blurb-html class=\\"blurb\\"><p>OpenStax partners have united with us to increase access to high-quality learning materials. Their low-cost tools integrate seamlessly with OpenStax books.</p></blurb-html></div><a class=\\"link-with-chevron\\" href=\\"http://openstax.org/partners\\">View Partners <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"chevron-right\\" class=\\"svg-inline--fa fa-chevron-right fa-w-10 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\"><path fill=\\"currentColor\\" d=\\"M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z\\"></path></svg></a></div></div></div>"`;
